selftests/job_api/test_features.py

```diff
@@ -0,0 +1,316 @@
+
+         ]})
+
+    suites.append(TestSuite.from_config(config_check_directory_exists))
+
+    # ========================================================================
+    # Test the content of a file
+    # ========================================================================
+    check_file_content = ('%s:%s.test_check_file_content'
+                          % (__file__, test_class))
+    config_check_file_content = (
+        {'run.references': [check_file_content],
+         'run.dict_variants': [
+
+             # finding the correct 'content' here is trick because any
+             # simple string is added to the variant file name and is
+             # found in the log file.
+             # Using DEBUG| makes the variant name have DEBUG_, working
+             # fine here.
+             {'namespace': 'job.output.loglevel',
+              'value': 'INFO',
+              'file': 'job.log',
+              'content': 'DEBUG| Test metadata',
+              'assert': False},
+
+             {'namespace': 'job.run.result.tap.include_logs',
+              'value': True,
+              'file': 'results.tap',
+              'content': "Command '/bin/true' finished with 0",
+              'assert': True},
+
+             {'namespace': 'job.run.result.tap.include_logs',
+              'value': False,
+              'file': 'results.tap',
+              'content': "Command '/bin/true' finished with 0",
+              'assert': False},
+
+             {'namespace': 'job.run.result.xunit.job_name',
+              'value': 'foo',
+              'file': 'results.xml',
+              'content': 'name="foo"',
+              'assert': True},
+
+             {'namespace': 'job.run.result.xunit.max_test_log_chars',
+              'value': 1,
+              'file': 'results.xml',
+              'content': '--[ CUT DUE TO XML PER TEST LIMIT ]--',
+              'assert': True,
+              'reference': ['/bin/false'],
+              'exit_code': 1},
+
+             {'namespace': 'run.failfast',
+              'value': True,
+              'file': 'results.json',
+              'content': '"skip": 1',
+              'assert': True,
+              'reference': ['/bin/false', '/bin/true'],
+              'exit_code': 9},
+
+             {'namespace': 'run.ignore_missing_references',
+              'value': 'on',
+              'file': 'results.json',
+              'content': '"pass": 1',
+              'assert': True,
+              'reference': ['/bin/true', 'foo'],
+              'exit_code': 0},
+
+         ]})
+
+    suites.append(TestSuite.from_config(config_check_file_content))
+
+    # ========================================================================
+    # Test if the result file was created
+    # ========================================================================
+    check_file_exists = ('%s:%s.test_check_file_exists'
+                         % (__file__, test_class))
+    config_check_file_exists = (
```

Comment:
One suggestion would be to add those dicts inside files for better organization. Let's say `tests/file_exists.json`, `tests/file_created.json`, ...

But I understand that we are running against the clock, so this could be a future reorg.

Reply:
@willianrampazzo please keep track of those comments for a future enhancement.

Reply:
Noted!